### PR TITLE
feat(NODE-4878): Add remaining log configurable client options

### PR DESCRIPTION
**Pull request by aditi-khare-mongoDB**
### Description
`mongodbLogComponentSeverities` and `mongodbLogMaxDocumentLength` are now supported options in client options through the MongoClient constructor. 

Users can now provide severity levels for loggable components through the `MongoClient` constructor, as well as max document length for their logs.

This feature is currently flagged as internal until [Logging](https://docs.google.com/document/d/1L3onvBqFaXuiN21sXMaFBRNS5kPnNYYckomt0s0ThmY/edit) is released.

#### What is changing?
The user can provide component severities through clientOptions. If environment variables as well as client options are present, the client options will take precedence. If certain components are not specified in either, their severity level will default to the value of 'default.' If default is not specified, the severity level for all components will be 'off.'

These client options are only supported for the constructor, not for connection string.

Example:
```
 new MongoClient('mongodb://localhost:27017', 
     {  mongodbLogComponentSeverities:  {
          client: 'trace',
          topology: 'trace',
          default: 'warn',
        },
        mongodbLogMaxDocumentLength: 800
    }
);
```

##### Is there new documentation needed for these changes?
Yes, documentation for `MongoClient` options should be updated with updated logging information. 

### Double check the following

- [ ] Ran `npm run check:lint` script
- [ ] Self-review completed using the [steps outlined here](https://github.com/mongodb/node-mongodb-native/blob/HEAD/CONTRIBUTING.md#reviewer-guidelines)
- [ ] PR title follows the [correct format](https://www.conventionalcommits.org/en/v1.0.0/): `type(NODE-xxxx)[!]: description`
  - Example: `feat(NODE-1234)!: rewriting everything in coffeescript`
- [ ] Changes are covered by tests
- [ ] New TODOs have a related JIRA ticket

### CODE REVIEW

File: test/unit/connection_string.test.ts [hunk 1]
```diff
@@ -883,4 +883,56 @@ describe('Connection String', function () {
       });
     });
   });
+
+  describe('non-genuine hosts', () => {
+    beforeEach(() => {
+      process.env.MONGODB_LOG_CLIENT = 'info';
+    });
+
```
Comment (aditi-khare-mongoDB) [inline test/unit/connection_string.test.ts:891]: Had to branch off of NODE-5452 in order to have client component, will rebase once that PR merges.

File: src/mongo_client.ts [hunk 2]
```diff
@@ -262,6 +267,16 @@ export interface MongoClientOptions extends BSONSerializeOptions, SupportedNodeC
...
+  /**
+   * @internal
+   * TODO: NODE-5671 - remove internal flag
+   */
+  mongodbLogComponentSeverities?: LogComponentSeveritiesClientOptions;
+  /**
+   * @internal
+   * TODO: NODE-5671 - remove internal flag
+   */
+  mongodbLogMaxDocumentLength?: string;
```
Comment (W-A-James) [inline src/mongo_client.ts:279]: Why have this set as a string instead of a number?

File: src/mongo_logger.ts [hunk 3]
```diff
@@ -522,22 +554,45 @@ export class MongoLogger {
...
+        client: resolveSeverityConfiguration(
+          combinedOptions.mongodbLogComponentSeverities?.client,
+          combinedOptions.MONGODB_LOG_CLIENT,
+          defaultSeverity
+        ),
         default: defaultSeverity
       },
       maxDocumentLength:
-        parseUnsignedInteger(combinedOptions.MONGODB_LOG_MAX_DOCUMENT_LENGTH) ?? 1000,
+        parseUnsignedInteger(combinedOptions.mongodblogMaxDocumentLength) ??
```
Comment (W-A-James) [inline src/mongo_logger.ts:593]:
```suggestion
        combinedOptions.mongodbLogMaxDocumentLength ??
```

We can just have the `mongodbLogMaxDocumentLength` be a number on the client options so we can do that validation inside `parseOptions`.

**Review by W-A-James — CHANGES_REQUESTED** [review]
Can we also add a unit test for the `mongodbLogMaxDocumentLength`?

File: src/mongo_client.ts (hunk 2 above)
Comment (aditi-khare-mongoDB) [reply src/mongo_client.ts:279]: It's set as string in the environment options interface [`MongoLoggerEnvOptions`](https://github.com/mongodb/node-mongodb-native/blob/fd58eec441d99493aff5c62bcc170c469efa679b/src/mongo_logger.ts#L124):
```MONGODB_LOG_MAX_DOCUMENT_LENGTH?: string;```

File: test/unit/mongo_client.test.js [hunk 4]
```diff
@@ -977,5 +977,27 @@ describe('MongoOptions', function () {
         });
       });
     });
+    context.only('when mongodbLogMaxDocumentLength is in options', function () {
```
Comment (W-A-James) [inline test/unit/mongo_client.test.js:980]: left-over .only

**Review by W-A-James — CHANGES_REQUESTED** [review]
Current changes look good. Approval pending changing `mongodbLogMaxDocumentLength` to `number` and the surrounding code to treat it as an integer specifically from outcome of Slack thread.

File: src/mongo_logger.ts [hunk 5]
```diff
@@ -126,10 +126,30 @@ export interface MongoLoggerEnvOptions {
...
+}
+
 /** @internal */
 export interface MongoLoggerMongoClientOptions {
   /** Destination for log messages */
   mongodbLogPath?: 'stdout' | 'stderr' | MongoDBLogWritable;
+  /** Severity levels for logger components */
+  mongodbLogComponentSeverities?: LogComponentSeveritiesClientOptions;
+  /** Max length of embedded EJSON docs. Setting to 0 disables truncation. Defaults to 1000. */
+  mongodblogMaxDocumentLength?: string;
```
[the remark this answers is not in the document]
Comment (aditi-khare-mongoDB) [reply src/mongo_logger.ts]: pushed changes!

File: src/mongo_logger.ts (hunk 3 above)
Comment (aditi-khare-mongoDB) [reply src/mongo_logger.ts]: pushed changes!

**Review by W-A-James — APPROVED** [review]
LGTM. Failures are unrelated

**Review by durran — APPROVED** [review]

**Pull request merged**